### ISSUE TITLE
SIGSEV in  decompileJUMP in decompile.c:1932

### ISSUE THREAD

**Issue by c1208828** [libming/libming#141]
https://docs.google.com/document/d/13xJhiIgDbqYwmR4j7aGEbKUU8KDl195mkw4rcvhT4J8/edit?usp=sharing
https://drive.google.com/open?id=1E7fkmjdvePRnsDoI1wpuZka0moHG7egl

(gdb) set args POC2
(gdb) r
header indicates a filesize of 453035351 but filesize is 60
<?php
$m = new SWFMovie(16);

ming_setscale(1.0);
$m->setRate(255.000000);
$m->setDimension(0, 0);
$m->setFrames(59397);
 Stream out of sync after parse of blocktype 12 (SWF_DOACTION). 32 but expecting 28.

/* SWF_DOACTION */

Program received signal SIGSEGV, Segmentation fault.
0x000000000043a1e9 in decompileJUMP (maxn=6, actions=0x691740, n=4) at decompile.c:1932
1932					if (sactif->Actions[sactif->numActions-1].SWF_ACTIONRECORD.ActionCode==SWFACTION_JUMP
(gdb) bt
#0  0x000000000043a1e9 in decompileJUMP (maxn=6, actions=0x691740, n=4) at decompile.c:1932
#1  decompileAction (n=4, actions=0x691740, maxn=6) at decompile.c:3325
#2  0x0000000000440a65 in decompileActions (indent=<optimized out>, actions=0x691740, n=6) at decompile.c:3494
#3  decompileSETTARGET (n=<optimized out>, actions=<optimized out>, maxn=<optimized out>, is_type2=<optimized out>)
    at decompile.c:3169
#4  0x000000000045752d in decompileActions (indent=<optimized out>, actions=0x6916a0, n=7) at decompile.c:3494
#5  decompile5Action (n=7, actions=0x6916a0, indent=indent@entry=0) at decompile.c:3517
#6  0x000000000040f34a in outputSWF_DOACTION (pblock=0x691250) at outputscript.c:1551
#7  0x000000000040211e in readMovie (f=0x690010) at main.c:281
#8  main (argc=<optimized out>, argv=<optimized out>) at main.c:354

(gdb) info all-registers 
rax            0x0	0
rbx            0x4	4
rcx            0x5	5
rdx            0x0	0
rsi            0xffffffffffffffb0	-80
rdi            0x0	0
rbp            0x6	0x6
rsp            0x7fffffffe1e0	0x7fffffffe1e0
r8             0x6918d0	6887632
r9             0x5	5
r10            0x5	5
r11            0x0	0
r12            0x691880	6887552
r13            0x140	320
r14            0x1	1
r15            0x5	5
rip            0x43a1e9	0x43a1e9 <decompileAction+90329>
eflags         0x10246	[ PF ZF IF RF ]
cs             0x33	51
ss             0x2b	43
ds             0x0	0
es             0x0	0
fs             0x0	0
gs             0x0	0
st0            0	(raw 0x00000000000000000000)
st1            0	(raw 0x00000000000000000000)
st2            0	(raw 0x00000000000000000000)
st3            0	(raw 0x00000000000000000000)
---Type <return> to continue, or q <return> to quit---
st4            0	(raw 0x00000000000000000000)
st5            0	(raw 0x00000000000000000000)
st6            0	(raw 0x00000000000000000000)
st7            0	(raw 0x00000000000000000000)
fctrl          0x37f	895
fstat          0x0	0
ftag           0xffff	65535
fiseg          0x0	0
fioff          0x0	0
foseg          0x0	0
fooff          0x0	0
fop            0x0	0
mxcsr          0x1f80	[ IM DM ZM OM UM PM ]

Breakpoint 1, decompileJUMP (maxn=6, actions=0x691740, n=4) at decompile.c:1932
1932					if (sactif->Actions[sactif->numActions-1].SWF_ACTIONRECORD.ActionCode==SWFACTION_JUMP
(gdb) l
1927		 
1928				if (OpCode(actions, n+i+j, maxn) == SWFACTION_IF)
1929				{
1930					sactif = (struct SWF_ACTIONIF *)&(actions[n+i+j]);
1931					/* chk whether last jump does lead us back to start of loop */
1932					if (sactif->Actions[sactif->numActions-1].SWF_ACTIONRECORD.ActionCode==SWFACTION_JUMP
1933					    && sactif->Actions[sactif->numActions-1].SWF_ACTIONJUMP.BranchOffset+
1934					    sactif->Actions[sactif->numActions-1].SWF_ACTIONJUMP.Offset==
1935					    actions[n].SWF_ACTIONRECORD.Offset )
1936					{

**Comment by FIOpwK** [comment]
assigned CVE:

> CVE-2018-9009 (https://nvd.nist.gov/vuln/detail/CVE-2018-9009):
  In libming 0.4.8, there is a use-after-free in the decompileJUMP function of
  the decompile.c file.

**Comment by hlef** [comment]
@FIOpwK No, this issue was assigned CVE-2018-11095. Why do you think this is a duplicate of #131 ?

@c1208828 Not reproducible on latest master. Seems to be fixed by 2027b24f403a859016a70bbdc79a8da1d6f128eb. Please, **do not ask for CVE numbers before checking for reproducibility on latest master**.

**Comment by hlef** [comment]
Alright, duplicate of #82.
